include/DDCEDViewer.h

```diff
@@ -181,7 +186,7 @@ CalorimeterDrawParams getCalorimeterParameters(dd4hep::Detector& theDetector, st
 
 //It suffices to perform the calculations in the first quadrant due to the detector's symmetry.
 //The signs of the tracks' directions are ultimately determined by the momenta.
-double calculateTrackLength(std::string type, dd4hep::Detector& theDetector, double x, double y, double z, double px, double py, double pz);
+double calculateTrackLength(std::string barrelName, std::string endcapName, dd4hep::Detector& theDetector, double x, double y, double z, double px, double py, double pz, double rel_X0 = 0.5);
```

Comment:
Is it in any way possible to easily add `rel_X0` to the `CalorimeterDrawParams` and determine it from the `barrelName` or the `endcapName`? It looks to be rather tightly connected to that and it would make it much clearer. Additionally, all the complexity of determining that value could be "hidden" in `getCalorimeterParameters`.